stdlib/public/Platform/android.modulemap

```diff
@@ -0,0 +1,676 @@
+// Swift directly, as these modules might be overriden by the C++ stdlib.
+// Instead, users should import 'Android' or another
+// umbrella module that includes these headers.
+
+// The top level 'Android' module can be included from Swift.
+module Android [system] {
+  header "SwiftAndroidNDK.h"
+  export *
+}
+
+// The top level 'Bionic' module is the C standard library
+// used in the Android NDK.
+module Bionic [system] {
+  header "SwiftBionic.h"
+  export *
+}
+
+// The individual modules for the Bionic C standard library used
+// by the Android NDK.
+module _assert [system] {
+  // <assert.h>'s use of NDEBUG requires textual inclusion.
+  textual header "assert.h"
+}
+module _complex [system] {
+  header "complex.h"
+  export *
+}
+module _ctype [system] {
+  header "ctype.h"
+  export *
+}
+module _errno [system] {
+  header "errno.h"
+  export *
+}
+module _fenv [system] {
+  header "fenv.h"
+  export *
+}
+// Use 'no_undeclared_includes' to avoid pulling in the libc++ headers.
+// FIXME: this can be dropped once NDK's libc++ 'std' module is split
+// into individual submodules.
+module _inttypes [system] [no_undeclared_includes] {
+  header "inttypes.h"
+  export *
+  use _stdint
+  use _sys_cdefs
+}
+module _limits [system] {
+  header "limits.h"
+  export *
+  explicit module posix_limits {
+    header "bits/posix_limits.h"
+    export *
+  }
+}
+module _locale [system] {
+  header "locale.h"
+  export *
+}
+module _malloc [system] {
+  header "malloc.h"
+  export *
+}
+module _math [system] {
+  header "math.h"
+  export *
+  link "m"
+}
+module _pty [system] {
+  header "pty.h"
+  export *
+}
+module _setjmp [system] {
+  header "setjmp.h"
+  export *
+}
+module _signal [system] {
+  header "signal.h"
+  // The 'signal.h' header unfortunately has a circular include
+  // with 'sys/ucontext.h' and thus it must be part of this module.
+  header "sys/ucontext.h"
+  export *
+  explicit module bits_signaltypes {
+    header "bits/signal_types.h"
+    export *
+  }
+}
+module _stdatomic [system] {
+  header "stdatomic.h"
+  export *
+}
+module _stdint [system] {
+  header "stdint.h"
+  export *
+}
+module _stdio [system] {
+  header "stdio.h"
+  export *
+  module stdio_ext {
+    header "stdio_ext.h"
+    export *
+  }
+}
+module _stdlib [system] {
+  header "stdlib.h"
+  export *
+}
+module _string [system] {
+  header "string.h"
+  export *
+}
+module _time [system] {
+  header "time.h"
+  export *
+  module sys_time {
+    header "sys/time.h"
+    export *
+  }
+  module sys_times {
+    header "sys/times.h"
+    export *
+  }
+}
+
+// POSIX and another android NDK headers.
+module alloca [system] {
+  header "alloca.h"
+  export *
+}
+
+module ar [system] {
+  header "ar.h"
+  export *
+}
+
+module cpio [system] {
+  header "cpio.h"
+  export *
+}
+
+module posix_filesystem [system] {
+  // This module groups all file, paths and filesystem
+  // operations into one module.
+  module dirent {
```

Comment:
I would personally make all of these submodules explicit. fcntl doesn't really imply you also want uio. But I guess there's also a balance with the convenience of a single import. I'm just thinking in Swift it's probably rare to have to dip down to posix unless you're actually writing the compiler, so you probably just want like `glob` or `libgen` and nothing else out of here.